packages/vscode-extension/src/webview/views/LaunchConfigurationView.tsx

```diff
@@ -0,0 +1,267 @@
+        update={update}
+      />
+
+      <div className="launch-configuration-section-margin" />
+
+      <Label>App Root</Label>
+      <AppRootConfiguration appRoot={appRoot} update={update} />
+      <div className="launch-configuration-section-margin" />
+
+      <Label>metro Config Path</Label>
+      <MetroConfigPathConfiguration metroConfigPath={metroConfigPath} update={update} />
+
+      <div className="launch-configuration-section-margin" />
+
+      <Label>is Expo</Label>
+      <IsExpoConfiguration isExpo={isExpo} update={update} />
+
+      <div className="launch-configuration-section-margin" />
+    </>
+  );
+}
+
+interface iosConfigurationProps {
+  scheme?: string;
+  configuration?: string;
+  update: <K extends keyof LaunchConfigurationOptions>(
+    key: K,
+    value: LaunchConfigurationOptions[K]
+  ) => void;
+  xcodeSchemes: string[];
+}
+
+function IosConfiguration({ scheme, configuration, update, xcodeSchemes }: iosConfigurationProps) {
+  const configurationInputRef = useRef<HTMLInputElement>(null);
+
+  const onSchemeChange = (newScheme: string) => {
+    if (newScheme === "Auto") {
+      update("ios", { scheme: undefined, configuration });
+      return;
+    }
+    update("ios", { scheme: newScheme, configuration });
+  };
+
+  const onConfigurationBlur = () => {
+    const newConfiguration = configurationInputRef.current?.value;
+    if (newConfiguration === "Auto" || !newConfiguration) {
+      update("ios", { scheme, configuration: "Auto" });
+      return;
+    }
+    update("ios", { scheme, configuration: newConfiguration });
+  };
+
+  const availableXcodeSchemes = xcodeSchemes.map((xcodeScheme) => {
+    return { value: xcodeScheme, label: xcodeScheme };
+  });
+
+  availableXcodeSchemes.push({ value: "Auto", label: "Auto" });
+
+  return (
+    <div className="container">
+      <div className="setting-description">Scheme:</div>
+      <Select
+        value={scheme ?? "Auto"}
+        onChange={onSchemeChange}
+        items={availableXcodeSchemes}
+        className="scheme"></Select>
+      <div className="setting-description">Configuration:</div>
+      <input
+        ref={configurationInputRef}
+        className="input-configuration"
+        type="string"
+        defaultValue={configuration ?? "Auto"}
+        onBlur={onConfigurationBlur}
+      />
+    </div>
+  );
+}
+
+interface androidConfigurationProps {
+  buildType?: string;
+  productFlavor?: string;
+  update: <K extends keyof LaunchConfigurationOptions>(
+    key: K,
+    value: LaunchConfigurationOptions[K]
+  ) => void;
+}
+
+function AndroidConfiguration({ buildType, productFlavor, update }: androidConfigurationProps) {
+  const buildTypeInputRef = useRef<HTMLInputElement>(null);
+  const productFlavorInputRef = useRef<HTMLInputElement>(null);
+
+  const onBuildTypeBlur = () => {
+    const newBuildType = buildTypeInputRef.current?.value;
+    if (newBuildType === "Auto" || !newBuildType) {
```

Comment:
First condition already takes care of undefined case.

Reply:
I am not concerned with undefined but with empty string here

Reply:
Can you please make it more explicit then?